File: src/main/java/com/nndmove/app/web/rest/vm/ManagedUserVM.java
package com.nndmove.app.web.rest.vm;

import com.nndmove.app.service.dto.AdminUserDTO;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * View Model extending the AdminUserDTO, which is meant to be used in the user management UI.
 */
@Getter
@Setter
@NoArgsConstructor
public class ManagedUserVM extends AdminUserDTO {

    public static final int PASSWORD_MIN_LENGTH = 4;

    public static final int PASSWORD_MAX_LENGTH = 100;

    @Size(min = PASSWORD_MIN_LENGTH, max = PASSWORD_MAX_LENGTH)
    private String password;

    @Override
    public String toString() {
        return "ManagedUserVM{" + super.toString() + "} ";
    }
}

File: src/main/java/com/nndmove/app/web/rest/vm/ManagedUserVM.kt
package com.nndmove.app.web.rest.vm

import com.nndmove.app.service.dto.AdminUserDTO
import jakarta.validation.constraints.Size

/**
 * View Model extending the AdminUserDTO, which is meant to be used in the user management UI.
 */
class ManagedUserVM : AdminUserDTO() {
    @JvmField
    var password: @Size(min = PASSWORD_MIN_LENGTH, max = PASSWORD_MAX_LENGTH) String? = null

    override fun toString(): String {
        return "ManagedUserVM{" + super.toString() + "} "
    }

    companion object {
        const val PASSWORD_MIN_LENGTH: Int = 4

        const val PASSWORD_MAX_LENGTH: Int = 100
    }
}
